packages/repo/src/block-map.ts

```diff
@@ -43,6 +44,24 @@ export class BlockMap {
       this.set(cid, bytes)
     })
   }
+
+  get size(): number {
+    return this.map.size
+  }
+
+  equals(other: BlockMap): boolean {
+    if (this.size !== other.size) {
+      return false
+    }
+    for (const entry of this.entries()) {
+      const otherBytes = other.get(entry.cid)
+      if (!otherBytes) return false
+      if (!uint8arrays.equals(entry.bytes, otherBytes)) {
+        return false
+      }
```

Comment:
At first glance it struck me that this check wasn't needed.  But I think I follow— is the idea that the block-map itself doesn't enforce how bytes map to cid, so its usage allows different bytes to be behind the same cid in two different block-maps (therefore we need to check the bytes in `equals()`)?

Reply:
yeah that was my thinking.

Another line of thought is that we prevent this from happening in the first place (with an integrity check on `set`). But I kind've like `BlockMap` being a relatively dumb class. Almost didn't want to put `add` on it (which serializes some data & calculates a CID), but it made all the consuming code so much simpler that I did.

Relatedly, I've actually realized that we don't enforce integrity checks correctly on our blockstore (I guess I'd assumed that the car reader does this, but it doesn't). So have been putting some thinking today into ensuring that blockstores have correct CID mappings